src/lib/expansion-panel/expansion-panel-core.ts

```diff
@@ -1,28 +1,55 @@
+import { frame } from '../core';
 import { IExpansionPanelAdapter } from './expansion-panel-adapter';
 import { ExpansionPanelAnimationType, ExpansionPanelOrientation, EXPANSION_PANEL_CONSTANTS } from './expansion-panel-constants';
 
 export interface IExpansionPanelCore {
   open: boolean;
   orientation: ExpansionPanelOrientation;
   animationType: ExpansionPanelAnimationType;
+  trigger: string;
+  triggerElement: HTMLElement | null;
   dispatchToggleEvent(): void;
 }
 
 export class ExpansionPanelCore implements IExpansionPanelCore {
   private _open = false;
   private _orientation: ExpansionPanelOrientation = 'vertical';
   private _animationType: ExpansionPanelAnimationType = 'default';
+  private _trigger = '';
 
   private _clickListener: EventListener = this._onClick.bind(this);
   private _keydownListener: EventListener = this._onKeydown.bind(this);
   private _animationCompleteListener = this._onAnimationComplete.bind(this);
+  private _slotListener = this._handleContentSlotChange.bind(this);
 
   constructor(private _adapter: IExpansionPanelAdapter) {}
 
-  public initialize(): void {
+  public async initialize(): Promise<void> {
     this._adapter.addHeaderListener('click', this._clickListener);
     this._adapter.addHeaderListener('keydown', this._keydownListener);
     this._adapter.setAnimationCompleteListener(this._animationCompleteListener);
+    this._adapter.addContentSlotListener(this._slotListener);
+    this._adapter.setContentId();
+    await frame();
+    this._syncTrigger();
+  }
+
+  public destroy(): void {
+    this._adapter.detachTriggerAria();
+    this._adapter.removeTriggerListeners();
+  }
+
+  private _handleContentSlotChange(): void {
+    this._adapter.setContentId();
+    this._syncTrigger();
+  }
+
+  private _syncTrigger(): void {
```

Comment:
We may want to handle the scenario where the trigger element is detached from the DOM dynamically and release our internal reference to it each time this method runs by checking if the trigger element ref is not connected.